common/util/cmd_positional_arguments.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_COMMON_UTIL_CMD_POSITIONAL_ARGUMENTS_H_
+#define VERIBLE_COMMON_UTIL_CMD_POSITIONAL_ARGUMENTS_H_
+
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+namespace verible {
+
+class CmdPositionalArguments {
+ public:
+  explicit CmdPositionalArguments(const std::vector<char *> &args)
+      : all_args_(args){};
+
+  // Main function that parses arguments.
+  absl::Status ParseArgs();
+
+  // Gets include directories arguments.
+  std::vector<absl::string_view> GetIncludeDirs();
+
+  // Gets macro defines arguments.
+  std::vector<std::pair<absl::string_view, absl::string_view>> GetDefines();
+
+  // Gets SV files arguments.
+  std::vector<absl::string_view> GetFiles();
```

Comment:
The first `const` is giving me a warning with clang-tidy:
`Clang-Tidy: Return type 'const TYPE' (aka 'const TYPE')
is 'const'-qualified at the top level,
which may reduce code readability without improving const correctness`

I think it means that it leads the reader to think the caller can't change the returned value, which is not true in general.

Reply:
I guess you forgot `&` after `const std::vector<...>`. This warnings appears only when returning const values/objects (which makes sense, because such `const` does nothing), not references to const value (which is used a lot to avoid forced copy).